### PR TITLE
[Google Cloud Platform Projects] Add X-Google-Api-Client as Header to identify Backstage use case for GCP telemetry

### PR DESCRIPTION
**Pull request by armandocomellas1**
> ## Hey, I just made a Pull Request!
> This PR adds a X-Google-Api-Client header to existing API requests in this package to clearly identify API requests from plugin Google Cloud Projects. headers are formatted wiith key value by the telemetry route plus the package version.
> 

> #### ✔️ Checklist
> * [x]  A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
> * [ ]  Added or updated documentation
> * [x]  Tests for new functionality and regression tests for bug fixes
> * [ ]  Screenshots attached (for UI changes)
> * [x]  All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Thank you for taking your time to review this PR.

### CODE REVIEW

File: plugins/gcp-projects/src/api/GcpClient.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import { GcpApi } from './GcpApi';
 import { Operation, Project } from './types';
 import { OAuthApi } from '@backstage/core-plugin-api';
+import packageinfo from '../../package.json';
```
Comment (Rugvip) [inline plugins/gcp-projects/src/api/GcpClient.ts:20]: Could you deconstruct the package info to only access the version here? It avoids bringing the entire package.json into the build. Same thing for the other similar solutions too tbh

Comment (armandocomellas1) [reply plugins/gcp-projects/src/api/GcpClient.ts]: Hi @Rugvip 

Sure, thank you for the annotation, l pushed the change with a new file named version.json in order to just get version number and not the entire package, let me know if that is what you asked for.

Thank you

File: plugins/gcp-projects/src/api/GcpClient.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import { GcpApi } from './GcpApi';
 import { Operation, Project } from './types';
 import { OAuthApi } from '@backstage/core-plugin-api';
+import packageinfo from '../../version.json';
```
Comment (Rugvip) [inline plugins/gcp-projects/src/api/GcpClient.ts:20]:
```suggestion
import { version } from '../../package.json';
```

This is what i had in mind

Comment (armandocomellas1) [reply plugins/gcp-projects/src/api/GcpClient.ts]: @Rugvip I tried this way, we have warnings linter errros

File: plugins/gcp-projects/src/api/GcpClient.ts (hunk 1 above)
Comment (armandocomellas1) [reply plugins/gcp-projects/src/api/GcpClient.ts]: @Rugvip in this way we not have the error that breaks the front end, but let me know if its ok from your point of view.

Thank you!

File: plugins/gcp-projects/src/api/GcpClient.ts (hunk 2 above)
Comment (Rugvip) [reply plugins/gcp-projects/src/api/GcpClient.ts]: Oh interesting, this is likely due to the new backend dev setup. Let's keep it as you had it originally, will have to look into this :+1:

Comment (armandocomellas1) [reply plugins/gcp-projects/src/api/GcpClient.ts]: @Rugvip Push it commit with the originally way to bring back the package version.

Thank you!

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit: !

**Pull request merged**